api/src/test/java/org/pmiops/workbench/billing/BillingProjectBufferServiceTest.java

```diff
@@ -164,4 +184,87 @@ public void fillBuffer_decreaseCapacity() {
         .createAllOfUsBillingProject(anyString());
   }
+
+  @Test
+  public void syncBillingProjectStatus_creating() {
+    billingProjectBufferService.bufferBillingProject();
+
+    ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+    verify(fireCloudService).createAllOfUsBillingProject(captor.capture());
+    String billingProjectName = captor.getValue();
+
+    BillingProjectStatus billingProjectStatus = new BillingProjectStatus();
+    billingProjectStatus.setCreationStatus(CreationStatusEnum.CREATING);
+    doReturn(billingProjectStatus).when(fireCloudService).getBillingProjectStatus(billingProjectName);
+    billingProjectBufferService.syncBillingProjectStatus();
+    assertThat(billingProjectBufferEntryDao.findByFireCloudProjectName(billingProjectName).getStatusEnum())
+        .isEqualTo(CREATING);
+  }
+
+  @Test
+  public void syncBillingProjectStatus_ready() {
+    billingProjectBufferService.bufferBillingProject();
+
+    ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+    verify(fireCloudService).createAllOfUsBillingProject(captor.capture());
+    String billingProjectName = captor.getValue();
+
+    BillingProjectStatus billingProjectStatus = new BillingProjectStatus();
+    billingProjectStatus.setCreationStatus(CreationStatusEnum.READY);
+    doReturn(billingProjectStatus).when(fireCloudService).getBillingProjectStatus(billingProjectName);
+    billingProjectBufferService.syncBillingProjectStatus();
+    assertThat(billingProjectBufferEntryDao.findByFireCloudProjectName(billingProjectName).getStatusEnum())
+        .isEqualTo(BillingProjectBufferStatus.AVAILABLE);
+  }
+
+  @Test
+  public void syncBillingProjectStatus_error() {
+    billingProjectBufferService.bufferBillingProject();
+
+    ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+    verify(fireCloudService).createAllOfUsBillingProject(captor.capture());
+    String billingProjectName = captor.getValue();
+
+    BillingProjectStatus billingProjectStatus = new BillingProjectStatus();
+    billingProjectStatus.setCreationStatus(CreationStatusEnum.ERROR);
+    doReturn(billingProjectStatus).when(fireCloudService).getBillingProjectStatus(billingProjectName);
+    billingProjectBufferService.syncBillingProjectStatus();
+    assertThat(billingProjectBufferEntryDao.findByFireCloudProjectName(billingProjectName).getStatusEnum())
+        .isEqualTo(ERROR);
+  }
+
+  @Test
+  public void syncBillingProjectStatus_multiple() {
+    billingProjectBufferService.bufferBillingProject();
+    billingProjectBufferService.bufferBillingProject();
+    billingProjectBufferService.bufferBillingProject();
+
+    ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+    verify(fireCloudService, times(3)).createAllOfUsBillingProject(captor.capture());
+    List<String> capturedProjectNames = captor.getAllValues();
+    doReturn(createBillingProjectStatus(CreationStatusEnum.READY)).when(fireCloudService)
```

Comment:
nit: loop?

Reply:
I decided to just write them out since the third mock return is different. It would end up being a n=2 loop + another statement which is just as verbose as writing out all three